### PR TITLE
Add package: capnproto

### PR DESCRIPTION
**Pull request by the-eater**

### CODE REVIEW

File: srcpkgs/capnproto/files/LICENSE [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Copyright (c) 2013-2014 Sandstorm Development Group, Inc. and contributors
```
Comment (Vaelatern) [inline srcpkgs/capnproto/files/LICENSE]: They really provide no copy of the license with their software?

File: srcpkgs/capnproto/template [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+pkgname=capnproto
+version=0.5.3
+revision=1
+wrksrc="${pkgname}-c++-${version}"
+build_style=gnu-configure
+depends="libcapnproto"
+short_desc="Cap'n Proto serialization/RPC system"
+maintainer="eater <hello@eaterofco.de>"
+license="MIT"
+homepage="https://capnproto.org/"
```
Comment (Vaelatern) [inline srcpkgs/capnproto/template]: Drop the end slash here

File: srcpkgs/capnproto/template [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+distfiles="${homepage}capnproto-c++-${version}.tar.gz"
```
Comment (Vaelatern) [inline srcpkgs/capnproto/template]: And add it here

File: srcpkgs/capnproto/template [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+checksum="cdb17c792493bdcd4a24bcd196eb09f70ee64c83a3eccb0bc6534ff560536afb"
+
+post_install() {
+	 vlicense "${FILESDIR}/LICENSE"
+}
+
+libcapnproto_package() {
+	short_desc+=" - shared libraries"
+	depends="libkj"
+
```
Comment (Vaelatern) [inline srcpkgs/capnproto/template]: No blank line here

File: srcpkgs/capnproto/template [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	pkg_install() {
+		vmove usr/include/capnp
+		vmove usr/lib/pkgconfig
+	}
+}
+
+libkj_package() {
+	short_desc+=" - utility shared libraries"
+
```
Comment (Vaelatern) [inline srcpkgs/capnproto/template]: You get the picture.

File: srcpkgs/capnproto/files/LICENSE (hunk 1 above)
Comment (the-eater) [inline srcpkgs/capnproto/files/LICENSE]: ah it was called `LICENSE.txt` while in the github repo they called it `LICENSE` fixed it

**Comment by the-eater** [comment]
Fixed commit message and set nocross to yes as it tries to run a binary it build just before in the same process :(

atm musl doesn't work, I will take a look tomorrow if I can make it work or not (I have no C/C++ exp.)

**Comment by Vaelatern** [comment]
https://www.gnu.org/software/gnulib/manual/html_node/execinfo_002eh.html

Not present on musl...

**Comment by the-eater** [comment]
@Vaelatern sorry for the late reaction, but yea I saw that, now included a patch which checks explicitly for GLIBC :)

File: srcpkgs/capnproto/template [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+wrksrc="${pkgname}-c++-${version}"
+build_style=gnu-configure
+depends="libcapnproto"
+short_desc="Cap'n Proto serialization/RPC system"
+maintainer="eater <hello@eaterofco.de>"
+license="MIT"
+homepage="https://capnproto.org"
+distfiles="${homepage}/capnproto-c++-${version}.tar.gz"
+checksum="cdb17c792493bdcd4a24bcd196eb09f70ee64c83a3eccb0bc6534ff560536afb"
+nocross=yes
```
Comment (Vaelatern) [inline srcpkgs/capnproto/template]: Why `nocross`?

**Comment by Vaelatern** [comment]
Okay, please rebase and solve the common/shlibs conflict (tip, go a few lines up from the bottom so additions merge smoother). Also please explain the nocross

File: srcpkgs/capnproto/template (hunk 6 above)
Comment (Vaelatern) [inline srcpkgs/capnproto/template]: Please set `nocross="relies on tools built at compilation to complete"` or something to that effect.

**Pull request closed without merging**